### PR TITLE
chore(contracts): deprecate PublicLock v8

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This removes support for PublicLock v8 from the codebase

# Issues

Fixes #
Refs https://github.com/unlock-protocol/unlock/issues/13445

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Review by julien51 — APPROVED** [review]

File: subgraph/src/unlock.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ export function handleNewLock(event: NewLock): void {
 
   // maxKeysPerAddress set to 1 prior to lock v10
   lock.maxKeysPerAddress = BigInt.fromI32(1)
-  if (version.ge(BigInt.fromI32(9))) {
+  if (version.ge(BigInt.fromI32(10))) {
```
Comment (julien51) [inline subgraph/src/unlock.ts:90]: Hum. That does not seem right. `ge` is greater or equal, right? If we we should keep it to 9!

File: subgraph/tests/locks-v8.test.ts [hunk 2]
```diff
@@ -1,108 +0,0 @@
-import {
```
Comment (julien51) [inline subgraph/tests/locks-v8.test.ts:1]: Let's keep this (but rename to v9) so we keep testing the logic.

File: subgraph/tests/locks.test.ts [hunk 3]
```diff
@@ -29,7 +29,6 @@ import {
   createPricingChangedEvent,
   createLockUpgradedEvent,
   createLockMetadata,
-  mockDataSourceV8,
```
Comment (julien51) [inline subgraph/tests/locks.test.ts:32]: Let's replace with `mockDataSourceV9` rather!

File: subgraph/tests/locks.test.ts [hunk 4]
```diff
@@ -57,7 +56,6 @@ import {
   baseTokenURI,
   maxNumberOfKeys,
   maxKeysPerAddress,
-  lockAddressV8,
```
Comment (julien51) [inline subgraph/tests/locks.test.ts:60]: same

File: subgraph/tests/locks.test.ts [hunk 5]
```diff
@@ -374,77 +372,3 @@ describe('Describe Locks events', () => {
     )
   })
 })
-
-describe('Describe Locks events (v8)', () => {
```
Comment (julien51) [inline subgraph/tests/locks.test.ts:378]: same

**Review by julien51 — COMMENTED** [review]
Please make sure we keep the v9 tests :)

**Comment by clemsos** [comment]
> Please make sure we keep the v9 tests :)

there were a bunch of cleanup to do in the subgraph tests to clarify what was happening with the role events. Should I move that to a separate PR ?

File: subgraph/src/public-lock.ts [hunk 6]
```diff
@@ -303,6 +301,7 @@ export function handleRoleGranted(event: RoleGrantedEvent): void {
         lock.lockManagers = [event.params.account]
       }
       lock.save()
+      log.debug('New lock manager', [event.params.account.toHexString()])
```
Comment (clemsos) [inline subgraph/src/public-lock.ts:304]:
```suggestion
```

File: subgraph/src/unlock.ts (hunk 1 above)
Comment (clemsos) [reply subgraph/src/unlock.ts:90]: yes it was a mistake in the previous subgraph code :/

File: subgraph/src/public-lock.ts [hunk 7]
```diff
@@ -341,13 +339,29 @@ export function handleRoleRevoked(event: RoleRevokedEvent): void {
       lock.keyGranters = newKeyGranters
       lock.save()
     }
+  } else if (
```
Comment (clemsos) [inline subgraph/src/public-lock.ts:342]: added the logic with OZ role, as we were still using our custom events in the subgraph

**Comment by julien51** [comment]
> there were a bunch of cleanup to do in the subgraph tests to clarify what was happening with the role events. Should I move that to a separate PR ? @julien51

That is a good idea indeed!

File: subgraph/src/public-lock.ts [hunk 8]
```diff
@@ -339,13 +339,29 @@ export function handleRoleRevoked(event: RoleRevokedEvent): void {
...
+    event.params.role.toHexString() ==
+    Bytes.fromHexString(LOCK_MANAGER).toHexString()
+  ) {
+    const lock = Lock.load(event.address.toHexString())
+    if (lock && lock.lockManagers) {
```
Comment (julien51) [inline subgraph/src/public-lock.ts:347]: Should we initialize `lock.lockManagers` to`[]` if it is not set?

**Comment by clemsos** [comment]
> > there were a bunch of cleanup to do in the subgraph tests to clarify what was happening with the role events. Should I move that to a separate PR ? @julien51
> 
> That is a good idea indeed!

the new PR is there https://github.com/unlock-protocol/unlock/pull/14570

**Pull request merged**